unity-renderer/Assets/DCLPlugins/RealmsPlugin/Tests/RealmsPluginTests.cs

```diff
@@ -0,0 +1,99 @@
+using System.Collections.Generic;
+using DCL;
+using DCL.Controllers;
+using Decentraland.Bff;
+using NSubstitute;
+using NUnit.Framework;
+using UnityEngine;
+using Variables.RealmsInfo;
+
+namespace DCLPlugins.RealmsPlugin
+{
+    public class RealmsPluginTests
+    {
+        private RealmsPlugin realmsPlugin;
+        private const string catalystRealmName = "CatalystRealmName";
+        private const string worldRealmName = "WorldRealmName";
+        private IRealmsModifier genericModifier;
+        private RealmsBlockerModifier realmsBlockerModiferSubstitute;
+        private ServiceLocator serviceLocator;
+
+        [SetUp]
+        public void SetUp()
+        {
+            // This is need to sue the TeleportController
+            serviceLocator = ServiceLocatorTestFactory.CreateMocked();
+            Environment.Setup(serviceLocator);
+
+            realmsPlugin = new RealmsPlugin();
+            realmsBlockerModiferSubstitute = Substitute.For<RealmsBlockerModifier>();
+            genericModifier = Substitute.For<IRealmsModifier>();
+            List<IRealmsModifier> substituteModifiers = new List<IRealmsModifier>() { realmsBlockerModiferSubstitute, genericModifier };
+            realmsPlugin.realmsModifiers = substituteModifiers;
+
+            SetCatalystRealmsInfo();
+        }
+        
+        [TearDown]
+        public void TearDown() { realmsPlugin.Dispose(); }
+
+        [TestCaseSource(nameof(GenericCases))]
+        public void ModifierCalledOnRealmChange(string realmName, bool isCatalist)
+        {
+            // Act
+            SetRealm(realmName);
+        
+            // Assert
+            genericModifier.Received().OnEnteredRealm(isCatalist, Arg.Any<AboutResponse >());
```

Comment:
i would specify the amount of calls. for example `genericModifier.Received(1)`

Reply:
Done!